pkg/segment/results/blockresults/runningstats.go

```diff
@@ -665,3 +679,176 @@ func (rj RunningStatsJSON) GetRunningStats() (runningStats, error) {
 	}
 	return rs, nil
 }
+
+func (rbr *RunningBucketResults) ToSerializedRunningBucketResults() *SerializedRunningBucketResults {
+	runningStats := make([]SerializedRunningStats, len(rbr.runningStats))
+	for i := 0; i < len(rbr.runningStats); i++ {
+		runningStats[i] = *rbr.runningStats[i].ToSerializedRunningStats()
+	}
+	groupedRunningStats := make(map[string][]SerializedRunningStats)
+	for k, v := range rbr.groupedRunningStats {
+		groupedRunningStats[k] = make([]SerializedRunningStats, len(v))
+		for i := 0; i < len(v); i++ {
+			groupedRunningStats[k][i] = *v[i].ToSerializedRunningStats()
+		}
+	}
+
+	return &SerializedRunningBucketResults{
+		RunningStats:        runningStats,
+		CurrStats:           rbr.currStats,
+		GroupedRunningStats: groupedRunningStats,
+		Count:               rbr.count,
+	}
+}
+
+func (rs *runningStats) ToSerializedRunningStats() *SerializedRunningStats {
+	return &SerializedRunningStats{
+		RawVal:    rs.rawVal,
+		Hll:       rs.hll,
+		RangeStat: rs.rangeStat,
+		AvgStat:   rs.avgStat,
+	}
+}
+
+func (srb *SerializedRunningBucketResults) ToRunningBucketResults(qid uint64) *RunningBucketResults {
+	runStats := make([]runningStats, len(srb.RunningStats))
+	for i := 0; i < len(srb.RunningStats); i++ {
+		runStats[i] = *srb.RunningStats[i].ToRunningStats()
+	}
+	groupedRunningStats := make(map[string][]runningStats)
+	for k, v := range srb.GroupedRunningStats {
+		groupedRunningStats[k] = make([]runningStats, len(v))
+		for i := 0; i < len(v); i++ {
+			groupedRunningStats[k][i] = *v[i].ToRunningStats()
+		}
+	}
+	return &RunningBucketResults{
+		runningStats:        runStats,
+		currStats:           srb.CurrStats,
+		groupedRunningStats: groupedRunningStats,
+		count:               srb.Count,
+		qid:                 qid,
+	}
+}
+
+func (srs *SerializedRunningStats) ToRunningStats() *runningStats {
+	return &runningStats{
+		rawVal:    srs.RawVal,
+		hll:       srs.Hll,
+		rangeStat: srs.RangeStat,
+		avgStat:   srs.AvgStat,
+	}
+}
+
+func GetRunningBucketResultsSliceForTest() []*RunningBucketResults {
```

Comment:
nit: this should probably be in a `..._test.go` file

Reply:
if I move this to a test file, I won't be able to call this function from `intermediateQueryResult_test.go`. I need the function in the same `pkg` so that it could also set the private fields.